kong/api/routes/upstreams.lua

```diff
@@ -40,7 +42,48 @@ return {
     end,
 
     GET = function(self, dao_factory)
-      crud.paginated_set(self, dao_factory.targets)
+      -- by default, just get a paginated list
+      if not self.params.active then
+        return crud.paginated_set(self, dao_factory.targets)
+      end
+
+      self.params.active = nil
```

Comment:
My first thought was with @thibaultcha proposal, but the @p0pr0ck5 explanation makes sense as well.

Googled around a bit and it seems the query parameter is the better option